core/src/test/java/org/apache/iceberg/TestRewriteFiles.java

```diff
@@ -323,14 +304,13 @@ public void testRewriteDataAndDeleteFiles() {
         statuses(DELETED, EXISTING));
 
     // We should only get the 5 manifests that this test is expected to add.
-    Assert.assertEquals("Only 5 manifests should exist", 5, listManifestFiles().size());
+    assertThat(listManifestFiles()).hasSize(5);
   }
 
-  @Test
+  @TestTemplate
   public void testRewriteDataAndAssignOldSequenceNumber() {
-    Assume.assumeTrue(
-        "Sequence number is only supported in iceberg format v2. ", formatVersion > 1);
-    Assert.assertEquals("Table should start empty", 0, listManifestFiles().size());
+    assumeThat(formatVersion).isGreaterThan(1);
```

Comment:
we should keep the description here so that it's clear why this is being skipped in certain situations

Reply:
Okay sure, add the description here.